core/templates/pages/admin-page/misc-tab/admin-misc-tab.directive.ts

```diff
@@ -161,6 +168,146 @@ angular.module('oppia').directive('adminMiscTab', [
+              var svgString = (
+                outputElement[0].getElementsByTagName('svg')[0].outerHTML);
+              var cleanedSvgString = (
+                ImageUploadHelperService.cleanMathExpressionSvgString(
+                  svgString));
+              var dimensions = (
+                ImageUploadHelperService.
+                  extractDimensionsFromMathExpressionSvgString(
+                    cleanedSvgString));
+              var dataURI = (
+                'data:image/svg+xml;base64,' + btoa(cleanedSvgString));
+              var invalidTagsAndAttributes = (
+                ImageUploadHelperService.getInvalidSvgTagsAndAttrs(dataURI));
+              var tags = invalidTagsAndAttributes.tags;
+              var attrs = invalidTagsAndAttributes.attrs;
+              if (tags.length === 0 && attrs.length === 0) {
+                var resampledFile = (
+                  ImageUploadHelperService.convertImageDataToImageFile(
+                    dataURI));
+                var date = new Date();
+                var now = date.getTime();
+                // This temporary Id will be used for adding and retrieving the
+                // raw image for each LaTeX string from the request body. For
+                // more details refer to the docstring in
+                // sendMathSvgsToBackend() in AdminBackendApiService.
+                var latexId = (
+                  now.toString(36).substr(2, 6) +
+                  Math.random().toString(36).substr(4));
+                resolve ({
+                  file: resampledFile,
+                  dimensions: {
+                    encoded_height_string: dimensions.height,
+                    encoded_width_string: dimensions.width,
+                    encoded_vertical_padding_string: dimensions.verticalPadding
+                  },
+                  latexId: latexId
+                });
+              } else {
+                AlertsService.addWarning(
+                  'SVG failed validation for LaTeX ' + 'inputLatexString');
+                reject();
+              }
+            });
+            // This will catch and log any error that occurs internally in
+            // MathJax.
+            MathJax.Hub.Register.MessageHook(
+              'Math Processing Error', function(message) {
+                LoggerService.error(message[2]);
+                reject();
+              });
+          });
+        };
+
+        var latexMapping = {};
+        var expIdToLatexMapping = null;
+        var numberOfLatexStrings = 0;
+        ctrl.generateSvgs = async function() {
+          var countOfSvgsGenerated = 0;
+          for (var expId in expIdToLatexMapping) {
+            var latexStrings = expIdToLatexMapping[expId];
+            latexMapping[expId] = {};
+            for (var i = 0; i < latexStrings.length; i++) {
+              var svgFile = await convertLatexToSvgFile(latexStrings[i]);
+              countOfSvgsGenerated++;
+              LoggerService.info(
```

Comment:
we cannot use `ctrl.setStatusMessage` here because that would require using a $rootscope.apply.
Hence I've used the loggerService.